lib/Github/Api/Repo.php

```diff
@@ -793,4 +793,30 @@ public function transfer($username, $repository, $newOwner, $teamId = [])
     {
         return $this->post('/repos/'.rawurldecode($username).'/'.rawurldecode($repository).'/transfer', ['new_owner' => $newOwner, 'team_id' => $teamId]);
     }
+
+    /**
+     * Create a repository using a template.
+     *
+     * @link https://developer.github.com/v3/repos/#create-a-repository-using-a-template
+     *
+     * @param string      $templateOwner
+     * @param string      $templateRepo
+     * @param string      $name
+     * @param string|null $owner
+     * @param string|null $description
+     * @param bool|null   $includeAllBranches
+     * @param bool|null   $private
+     *
+     * @return array
+     */
+    public function createFromTemplate($templateOwner, $templateRepo, $name, $owner = null, $description = null, $includeAllBranches = false, $private = null)
```

Comment:
Can you replace all the "parameter" variables with just a `$params/$parameters` array and let the user provide the array of data they want to send.

This will make it easier to adapt changes from github to the allowed options (removed and added options)

Can you also add the appropriate typehints to the remaining parameters? (this will also allow us to drop the superfluous `@param` docblocks!